zcash_client_sqlite/src/wallet/init/migrations/ensure_default_transparent_address.rs

```diff
@@ -0,0 +1,50 @@
+//! Ensures that an external transparent address exists in the `addresses` table for each
+//! non-hardened child index starting at index 0 and ending at the index corresponding to default
+//! address for the account.
+
+use std::collections::HashSet;
+use uuid::Uuid;
+
+use rusqlite::Transaction;
+use schemerz_rusqlite::RusqliteMigration;
+use zcash_protocol::consensus;
+
+use super::transparent_gap_limit_handling;
+use crate::wallet::init::WalletMigrationError;
+
+pub(super) const MIGRATION_ID: Uuid = Uuid::from_u128(0x702cf97b_8395_4edc_b584_5c9f87f0ef35);
+
+const DEPENDENCIES: &[Uuid] = &[transparent_gap_limit_handling::MIGRATION_ID];
+
+pub(super) struct Migration<P> {
+    pub(super) params: P,
+}
+
+impl<P> schemerz::Migration<Uuid> for Migration<P> {
+    fn id(&self) -> Uuid {
+        MIGRATION_ID
+    }
+
+    fn dependencies(&self) -> HashSet<Uuid> {
+        DEPENDENCIES.iter().copied().collect()
+    }
+
+    fn description(&self) -> &'static str {
+        "Add support for general transparent gap limit handling, unifying the `addresses` and `ephemeral_addresses` tables."
+    }
+}
+
+impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
+    type Error = WalletMigrationError;
+
+    fn up(&self, conn: &Transaction) -> Result<(), WalletMigrationError> {
+        #[cfg(feature = "transparent-inputs")]
+        transparent_gap_limit_handling::insert_initial_transparent_addrs(conn, &self.params)?;
```

Comment:
The code in the `transparent_gap_limit_handling` migration that updates `transparent_received_outputs.address_id` is dependent on the set of cached transparent addresses. Does it need to be re-run here?

Reply:
No; if the `transparent_gap_limit_handling` migration ran successfully, then all of the `transparent_received_outputs` entries are correctly updated with the associated address ids, and we will not have inserted any records into `transparent_received_outputs` that could violate the relationship to the `addresses` table after that exactly because `address_id` is required to be non-null.